### ISSUE TITLE
Poor readability in BNFs of the AST documentation

### ISSUE THREAD

**Issue by morioprog** [dart-lang/sdk#47440]
## Problem

Currently, BNFs in some documents in [the ast library](https://pub.dev/documentation/analyzer/latest/dart_ast_ast/dart_ast_ast-library.html) are hard to read:

1. Broken line breaks.
    - [GenericFunctionType class](https://pub.dev/documentation/analyzer/latest/dart_ast_ast/GenericFunctionType-class.html)
2. The correspondence of quotes is hard to parse.
    - [SwitchStatement class](https://pub.dev/documentation/analyzer/latest/dart_ast_ast/SwitchStatement-class.html)
    - [ClassDeclaration class](https://pub.dev/documentation/analyzer/latest/dart_ast_ast/ClassDeclaration-class.html)

## Suggestion

1. Fix line breaks as in this class: [ConstructorDeclaration class](https://pub.dev/documentation/analyzer/latest/dart_ast_ast/ConstructorDeclaration-class.html).
2. Use the monospaced font in BNFs to make it easier to read.

**Comment by srawlins** [comment]
Largely a duplicate of https://github.com/dart-lang/sdk/issues/24544
